src/ngraph/ngraph_sgcompiler.cc

```diff
@@ -342,9 +340,10 @@ void SGCompiler::CompileSubgraph(std::shared_ptr<Graph> sub_graph) {
   }
 
   if (exe_mode_ == GraphExeMode::kTrain) {
-    ngraph::FpropCache fprop_cache;
-    fprop_cache.node_param_map = std::make_shared<ngraph::NodeMap>();
-    CompileForwardBackward(sub_graph, f, maybe_bf, exe_mode_, fprop_cache);
+    sub_graph->fprop_cache->fprop = f;
+    sub_graph->fprop_cache->bprop = maybe_bf;
+    sub_graph->fprop_cache->node_param_map =
+        std::make_shared<ngraph::NodeMap>();
```

Comment:
Is it possible for `sub_graph->fprop_cache` to ever be null when this code runs?

Reply:
Yes! It didn't used to be, but it now is. Will fix.